src/pokemon.c

```diff
@@ -3530,7 +3530,14 @@ bool8 PokemonUseItemEffects(struct Pokemon *mon, u16 item, u8 partyIndex, u8 mov
                 {
                     u16 species = GetMonData(mon, MON_DATA_SPECIES, NULL);
                     dataUnsigned = sExpCandyExperienceTable[param - 1] + GetMonData(mon, MON_DATA_EXP, NULL);
-                    if (dataUnsigned > gExperienceTables[gSpeciesInfo[species].growthRate][MAX_LEVEL])
+
+                    if (B_RARE_CANDY_CAP && EXP_CAP_HARD)
+                    {
+                        u32 currentLevelCap = GetCurrentLevelCap();
+                        if (dataUnsigned > gExperienceTables[gSpeciesInfo[species].growthRate][currentLevelCap])
+                            dataUnsigned = gExperienceTables[gSpeciesInfo[species].growthRate][currentLevelCap];
```

Comment:
I wonder if we need to build a safeguard here to prevent `currentLevelCap` being larger than `MAX_LEVEL` to prevent stuff from being read OoB. This also goes for the experience table check in the other file

Reply:
Should I add something that caps it at `MAX_LEVEL`?

Reply:
Potentially, but maybe that's overly zealous. Just wanted to put it on the table

Reply:
I've tested it and it never goes beyond level 100.